convert2rhel/actions/system_checks/check_firewalld_availability.py

```diff
@@ -0,0 +1,133 @@
+from convert2rhel.systeminfo import system_info
+
+
+logger = logging.getLogger(__name__)
+
+# Path to the firewalld config file.
+FIREWALLD_CONFIG_FILE = "/etc/firewalld/firewalld.conf"
+
+# Regex that identify if CleanupModulesOnExit is set to either yes or true
+# considering blank spaces in between the value and case insensitive.
+CLEANUP_MODULES_ON_EXIT_REGEX = re.compile(r"(?i)CleanupModulesOnExit\s*=\s*(yes|true)")
+
+
+def _is_modules_cleanup_enabled():
+    """Verify firewalld modules cleanup config is enabled.
+
+    :returns: Whether or not the CleanupModulesOnExit is set to true in
+        firewalld config.
+    :rtype: bool
+    """
+    # Return false if the config file does not exist. Either it means that
+    # firewalld is not installed or the config file was removed.
+    if not os.path.exists(FIREWALLD_CONFIG_FILE):
+        logger.debug("%s does not exist." % FIREWALLD_CONFIG_FILE)
+        return False
+
+    contents = []
+    with open(FIREWALLD_CONFIG_FILE, mode="r") as handler:
+        contents = [line.strip() for line in handler.readlines() if line.strip()]
+
+    # Contents list is empty for some reason, better to assume that there
+    # is no content in the file that was read.
+    if not contents:
+        logger.debug("%s is empty." % FIREWALLD_CONFIG_FILE)
+        return True
+
+    # If the CleanupModulesOnExit is not present inside the contents list, we
+    # can return True since the default behavior for firewalld is to consider
+    # CleanupModulesOnExit as true. Ignoring # and ; as they are ignored in the
+    # config parser for firewalld as well:
+    # https://github.com/firewalld/firewalld/blob/46d54dcbdff94423686d67befc78ca8d601fce60/src/firewall/core/io/firewalld_conf.py#L85
+    option_present = any(
+        item.strip().startswith("CleanupModulesOnExit") for item in contents if not item.startswith(("#", ";"))
+    )
+    if not option_present:
+        logger.debug(
+            "Couldn't find CleanupModulesOnExit in firewalld.conf. Treating it as enabled because of default behavior."
+        )
+        return True
+
+    # If the config file has this option set to true/yes, then we need to
+    # return True to ask the user to change it to False.
+    if list(filter(CLEANUP_MODULES_ON_EXIT_REGEX.match, contents)):
+        logger.debug("CleanupModulesOnExit option enabled in %s" % FIREWALLD_CONFIG_FILE)
+        return True
+
+    # Default to return False as it is possible that the CleanupModulesOnExit
+    # is set to no in the config already.
+    return False
+
+
+class CheckFirewalldAvailability(actions.Action):
+    id = "CHECK_FIREWALLD_AVAILABILITY"
+
+    def run(self):
+        """Error out if the firewalld service is running on the system."""
+        super(CheckFirewalldAvailability, self).run()
+        logger.task("Prepare: Check that firewalld is running")
+
+        if system_info.id != "oracle" and system_info.version.major != 8 and system_info.version.minor < 8:
```

Comment:
I'm not sure this condition works. The inhibitor happens also on OL7:
http://artifacts.osci.redhat.com/testing-farm/6c246209-3b6d-4f67-82cc-17aee0fd3f5c/

I noticed this on CentOS7:
```
[2023-10-30T07:43:47+0000] TASK - [Prepare: Check that firewalld is running] *************************
[2023-10-30T07:43:47+0000] DEBUG - Calling command '/usr/bin/systemctl show -p ActiveState firewalld'
Firewalld service reported that it is not running.
CHECK_FIREWALLD_AVAILABILITY has succeeded
```
the centos VMs don't have the `firewalld` package at all I believe. Anyway, it seems that it wasn't skipped for some reason

Reply:
Seems that it is fixed now, thanks @r0x0d!:

- http://artifacts.osci.redhat.com/testing-farm/73014ac9-4a94-4595-b8a8-21fdb72f84ce/work-activation_key8644_n6m/log.txt
- http://artifacts.osci.redhat.com/testing-farm/ed0c2fa5-0980-4160-a737-82385e5b6707/work-activation_keyenr_ngna/log.txt
- http://artifacts.osci.redhat.com/testing-farm/f7139855-6998-4b38-995c-92d38cd9361f/work-rhsm8cbya2nf/log.txt